src/settings/autostartmanager/abstractautostartmanager.cpp

```diff
@@ -38,6 +40,8 @@ AbstractAutostartManager *AbstractAutostartManager::createAutostartManager(QObje
     if (PortalAutostartManager::isAvailable())
         return new PortalAutostartManager(parent);
     return new UnixAutostartManager(parent);
+#elif defined(Q_OS_DARWIN)
+    return new macOSAutostartManager(parent);
```

Comment:
Maybe put `m_autostartManager` in settingsdialog.h under the define instead?

Reply:
Hmm, I think the logic for implementation selection fits better to the base class, but I'm not familiar with C++ best practices in this regard.

Reply:
You are right, generally it's better to use a separate class. But since we don't have a working implementation I would prefer to put a stub via define around `m_autostartManager` inside `settingsdialog.h` to not construct it at all.

Reply:
The implementation is very much doable, at least a cheap-and-dirty one (using `system("osascript ...")`). I just didn't have time for it yes, so created a dummy manager class.

Reply:
If you want to go with a temporary stub, I would prefer to have several ifdefs. But if you provide the working implementation - it would be awesome and I would go with `MacOSAutostartManager`, yes.

Reply:
Ok, here is the implementation largely copied from KeePassXC. I hope it works wll enough (I've only checked that it creates and deletes the `.plist` file as expected).